src/reducers/join.ts

```diff
@@ -0,0 +1,54 @@
+import _ from 'lodash'
+import existingThoughtChange from './existingThoughtChange'
+import { getAllChildren, getChildPath, getNextRank, getPrevRank } from '../selectors'
+import { SimplePath } from '../types'
+import { head, parentOf, pathToContext, reducerFlow } from '../util'
+import { State } from '../util/initialState'
+import existingThoughtMove from './existingThoughtMove'
+import existingThoughtDelete from './existingThoughtDelete'
+
+/** Join two or more thoughts split by spaces. */
+const join = (state: State) => {
+
+  const { cursor } = state
+
+  if (!cursor) return state
+
+  const contextRanked = pathToContext(parentOf(cursor))
```

Comment:
Since `pathToContext` converts the `Path` to a `Context`, it's actually unranked. `context` would be a better name.